cloud/scope/machine_pool.go

```diff
@@ -536,27 +596,24 @@ func (m *MachinePoolScope) CreateInstancePool(ctx context.Context) (*core.Instan
 }
 
 // UpdatePool attempts to update the instance pool
-func (m *MachinePoolScope) UpdatePool(ctx context.Context, instancePool *core.InstancePool) error {
-
+func (m *MachinePoolScope) UpdatePool(ctx context.Context, instancePool *core.InstancePool) (*core.InstancePool, error) {
 	if instancePoolNeedsUpdates(m, instancePool) {
-		m.Info("updating instance pool")
-
-		replicas := int(1)
-		if m.MachinePool.Spec.Replicas != nil {
-			replicas = int(*m.MachinePool.Spec.Replicas)
-		}
-
+		m.Info("Updating instance pool")
+		replicas := int(*m.MachinePool.Spec.Replicas)
 		req := core.UpdateInstancePoolRequest{InstancePoolId: instancePool.Id,
 			UpdateInstancePoolDetails: core.UpdateInstancePoolDetails{
-				Size: common.Int(replicas),
-			}}
-
-		if _, err := m.ComputeManagementClient.UpdateInstancePool(ctx, req); err != nil {
-			return errors.Wrap(err, "unable to update instance pool")
+				Size:                    common.Int(replicas),
+				InstanceConfigurationId: m.OCIMachinePool.Spec.InstanceConfiguration.InstanceConfigurationId,
+			},
 		}
+		resp, err := m.ComputeManagementClient.UpdateInstancePool(ctx, req)
+		if err != nil {
+			return nil, err
```

Comment:
If it fails to update, I don't recall, is the error message clear enough that update failed? I think that is why the extra `Wrap` was used. 🤷

Reply:
fixed